server/web_ui/shared/src/models.rs

```diff
@@ -1,12 +1,101 @@
-use gloo::storage::{
-    // LocalStorage as PersistentStorage,
-    SessionStorage as TemporaryStorage,
-    Storage,
-};
+#[cfg(debug_assertions)]
+use gloo::console;
+use gloo::storage::{LocalStorage, SessionStorage as TemporaryStorage, Storage};
+use kanidm_proto::oauth2::AuthorisationRequest;
 use kanidm_proto::v1::{CUSessionToken, CUStatus};
+use wasm_bindgen::UnwrapThrowExt;
+
+pub fn set_bearer_token(r: String) {
+    LocalStorage::set("bearer_token", r).expect_throw("failed to set bearer_token");
+}
+
+pub fn get_bearer_token() -> Option<String> {
+    let l: Result<String, _> = LocalStorage::get("bearer_token");
+    #[cfg(debug_assertions)]
+    console::debug!(format!(
+        "login_hint::get_login_remember_me -> present={:?}",
+        l.is_ok()
+    )
+    .as_str());
+    l.ok()
+}
+
+pub fn clear_bearer_token() {
+    LocalStorage::delete("bearer_token");
+}
 
 /// Pulls the "cred_update_session" element from the browser's temporary storage
 pub fn get_cred_update_session() -> Option<(CUSessionToken, CUStatus)> {
     let l: Result<(CUSessionToken, CUStatus), _> = TemporaryStorage::get("cred_update_session");
     l.ok()
 }
+
+pub fn push_return_location(l: &str) {
+    TemporaryStorage::set("return_location", l)
+        .expect_throw("failed to set return_location in temporary storage");
+}
+
+pub fn pop_return_location() -> String {
```

Comment:
Wasn't this an enum in the past? Should we consider similar here?

Reply:
Yeah this is just me messin' around while splitting it up and trying to make things go. This ... is a messy part that `yew-rs`'s routers might cause some friction with, since it's an SPA-framework and expect some things, but I have ideas.

Reply:
Turns out to get it working it has to be a string value because they don't share routers/enums, nor do we want them to.